app/javascript/gobierto_data/webapp/components/sets/data/SQLEditorHeader.vue

```diff
@@ -122,31 +128,48 @@ export default {
     },
     queryName: {
       type: String,
-      default: null,
+      default: '',
     },
+    enabledSavedButton: {
+      type: Boolean,
+      default: false
+    },
+    showRevertQuery: {
+      type: Boolean,
+      default: false
+    },
+    showPrivate: {
+      type: Boolean,
+      default: false
+    },
+    isQuerySaved: {
+      type: Boolean,
+      default: false
+    }
   },
   data() {
     return {
       labelRecents: I18n.t("gobierto_data.projects.recents") || "",
       labelQueries: I18n.t("gobierto_data.projects.queries") || "",
       labelRunQuery: I18n.t("gobierto_data.projects.runQuery") || "",
+      labelResetQuery: I18n.t("gobierto_data.projects.resetQuery") || "",
+      labelSave: I18n.t("gobierto_data.projects.save") || "",
       labelQueryName: I18n.t("gobierto_data.projects.queryName") || "",
-      labelModifiedQuery: I18n.t("gobierto_data.projects.modifiedQuery") || "",
       labelButtonQueries: I18n.t("gobierto_data.projects.buttonQueries") || "",
       labelButtonRecentQueries:
         I18n.t("gobierto_data.projects.buttonRecentQueries") || "",
       labelButtonRunQuery:
         I18n.t("gobierto_data.projects.buttonRunQuery") || "",
       removeLabelBtn: false,
       isQueriesModalActive: false,
-      isRecentModalActive: false,
+      isRecentModalActive: false
     };
   },
   created() {
     // it has to be the same event (keydown) as SQLEditorCode
     document.addEventListener("keydown", this.keyboardShortcutsListener);
   },
-  beforeDestroy() {
+  deactivated() {
```

Comment:
Rollback to destroy

Reply:
if we use `beforeDestroy` instead of  `deactivated` this bug appears again: https://github.com/PopulateTools/issues/issues/1013

Reply:
If #1013 happens, that means `SQLEditorHeader` is not being destroyed. And that happens because `SQLEditorHeader` is wrapped in a `<keep-alive>` clause. Is that right?
Then, we've to look why is SQLEditorHeader kept alive. It shouldn't. It only belongs to DataTab component, who is in charge to create it. Is DataTab kept alive also? Why reason?

Reply:
The point of this is about the lifecycles. `created` <-> `destroyed`. `activated` <-> `deactivated`. But never mixed.